src/openforms/submissions/admin_views.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth.mixins import LoginRequiredMixin, PermissionRequiredMixin
+from django.views.generic import ListView
+
+from .models import Submission
+
+
+class LogsEvaluatedLogicView(PermissionRequiredMixin, LoginRequiredMixin, ListView):
+
+    template_name = "submission_logs/submission_logs.html"
+    context_object_name = "logs_activity"
+    permission_required = "submissions.view_submission"
+
+    def get_queryset(self):
+        self.id = self.kwargs["submission_id"]
+        self.submission = Submission.objects.get(id=self.id)
+        self.queryset = self.submission.logs.filter(
+            template="logging/events/submission_logic_evaluated.txt"
+        ).order_by("timestamp")
+        return super().get_queryset()
+
+    def get_context_data(self):
+        context = super().get_context_data()
+        context["opts"] = Submission._meta
+        context["original"] = self.submission
+        return context
```

Comment:
Couple improvements to be made here, I'll start with the anti-patterns first:

1. you typically shouldn't assign anything to `self` outside of the `__init__` method. (the `self.id = ...` doesn't really make much sense, especially because the data is already available as `self.kwargs["submission_id"]`. In its current form, you're sort of "polluting the namespace". A local variable would've been sufficient
2. assigning `self.queryset = ...` suffers from the problem above, but it's also not really how you're supposed to implement this. Django has a default implementation for `get_queryset` that:
    - either checks if `queryset` is set, if it is -> `return self.queryset.all()`
    - if no queryset is set, check if `model` is set and get the queryset from that: `return self.model.objects.all()`

    You can override this by just returning a queryset from the `get_queryset` method and django will be fine with that - you definitely don't need to overwrite the instance attribute.

Then, some further improvements:

1. You can make the `self.submission` thing a bit more self-contained by implementing it as a property with an internal cache. That makes it so you do the database lookup only once, without doing too much `self.foo = bar` stuff - it's a bit of a trade-off. In my version, it is at least made "obvious" that this is some internal caching API by using the `_` prefix.
2. Since this view is used in the admin, you typically want to enforce that it's an admin user viewing it, on top of having the required permission. We have an existing mixin for that in `openforms.utils.mixins.UserIsStaffMixin`. Your tests should also reflect this in the form of setting up a non-staff user with the permission - they should not have access to this view!

All in all, I'd rewrite this as:

```py
from openforms.utils.mixins import UserIsStaffMixin
from django.shortcuts import get_object_or_404


class LogsEvaluatedLogicView(UserIsStaffMixin, PermissionRequiredMixin, ListView):

    template_name = "submission_logs/submission_logs.html"
    context_object_name = "logs_activity"
    permission_required = "submissions.view_submission"

    @property
    def submission(self):
        if not hasattr(self, "_submission"):
            self._submission = get_object_or_404(Submission, id=self.kwargs["submission_id"])
        return self._submission

    def get_queryset(self):
        logic_log_template = "logging/events/submission_logic_evaluated.txt"
        queryset = self.submission.logs.filter(template=logic_log_template)
        return queryset.order_by("timestamp")

    def get_context_data(self):
        context = super().get_context_data()
        context.update({
            "opts": Submission._meta,
            "original": self.submission,
        })
        return context
```